examples/dashboard_embeddable_examples/public/by_value/input_editor.tsx

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { EuiButton, EuiCodeEditor } from '@elastic/eui';
+
+// @ts-ignore
+const TextMode = window.ace.acequire('ace/mode/json').Mode;
+export const InputEditor = <T,>(props: { input: T; onSubmit: (value: T) => void }) => {
+  const input = JSON.stringify(props.input, null, 4);
+  const [value, setValue] = React.useState(input);
+  const isValid = (() => {
+    try {
+      JSON.parse(value);
+      return true;
+    } catch (e) {
+      return false;
+    }
+  })();
+  React.useEffect(() => {
+    setValue(input);
+  }, [input]);
+  return (
+    <>
+      <EuiCodeEditor
+        mode={new TextMode()}
```

Comment:
I peeked at the [source](https://github.com/elastic/kibana/blob/master/src/plugins/es_ui_shared/public/components/json_editor/use_json.ts) and it looks like its benefits are that it wraps EuiCodeEditor in a EuiFormRow and it bakes in some EuiCodeEditor settings, so you get some cross-Kibana consistency wherever this component is used. To be fair I believe we're only using this component in one place right now. @sebelga Is there anything else?

Reply:
I totally missed this ping! 😄 

The other benefit is that it can be controlled or uncontrolled. If it is not controlled, it has its own internal state and validates the JSON inside a denounce (300ms) to avoid validating the Json (`JSON.parse`) on each keystroke.

And as @cjcenizal mentioned, it gives us a consistent look on all our JSON editor (even though props can be passed to override the default settings.